File: 2021_Aug/12.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        resKey = []
        res = []
        for string in strs:
            tmpKey = {}
            for alphabet in string:
                if (alphabet in tmpKey):
                    tmpKey[alphabet] += 1
                else:
                    tmpKey[alphabet] = 1
            if (tmpKey in resKey):
                res[resKey.index(tmpKey)].append(string)
            else:
                resKey.append(tmpKey)
                res.append([string])
        return res

File: 2021_April/12.cpp
class Solution {
public:
    vector<int> constructArray(int n, int k) {
        vector<int> res;
        int i = 1, j = n;
        while (i <= j) {
            if (k > 1) res.push_back(k-- % 2 ? i++ : j--);
            else res.push_back(i++);
        }
        return res;
    }
};